functions/src/utils/agora.js

```diff
@@ -0,0 +1,41 @@
+const functions = require("firebase-functions");
+
+const { RtcTokenBuilder, RtcRole } = require("agora-access-token");
+
+const AGORA_CONFIG = functions.config().agora;
+
+const appId = AGORA_CONFIG.app_id;
+const appCertificate = AGORA_CONFIG.app_certificate;
+
+// @debt should we move this into AGORA_CONFIG, or maybe venue config in firestore or similar?
+const expirationTimeInSeconds = 3600;
+
+const getExpirationTime = () => {
+  const currentTimestamp = Math.floor(Date.now() / 1000);
+
+  return currentTimestamp + expirationTimeInSeconds;
+};
+
+const generateAgoraTokenForAccount = ({ channelName, account, role }) => {
+  // @debt we should enforce a stricter security requirement on channelName. Maybe use UUIDs?
```

Comment:
UUID is the best option. But right now it's ok to have just a string

Reply:
For continuity, see also this thread https://github.com/sparkletown/sparkle/pull/1673#discussion_r659323441 on the same topic